src/java_bytecode/java_types.h

```diff
@@ -393,6 +397,46 @@ to_java_implicitly_generic_class_type(java_class_typet &type)
   return static_cast<java_implicitly_generic_class_typet &>(type);
 }
 
+/// \param type The type to check
+/// \return True if the given type is a generic class type or an implicitly
+/// generic class type, false otherwise
+inline const bool is_java_any_generic_class_type(const typet &type)
+{
+  const bool value = is_java_generic_class_type(type) ||
+                     is_java_implicitly_generic_class_type(type);
+  return value;
+}
+
+/// \param type Source type
+/// \return The vector of implicitly generic and generic type parameters of
+/// the given type.
+inline const std::vector<java_generic_parametert> get_all_generic_parameters
```

Comment:
Why are these methods marked as `inline`?

Reply:
The functions are reasonably small (at least the first one) and may end up being called quite often, I thought. Would you remove the `inline`?

Reply:
In C++ inline no longer means inline this function. It is, at best, a hint to the compiler that this might be a good candidate to inline. 

Modern compilers will ignore this as they (almost) always know better and will inline according to their own criteria (including functions that are not marked inline). Instead, it is mainly for AFAIK allowing [more than one definition](http://en.cppreference.com/w/cpp/language/inline).

Reply:
I see, thanks for the link. I will remove the `inline` then.

Reply:
Seems that to keep the functions in the header (which I think makes sense, since similar functions for java types are in the header too) I have to keep the `inline`. Otherwise I get errors about multiple definitions.

Reply:
Since non-trivial I would favour in the cpp file, but just preference

Reply:
Since I only used this function while updating the generic map and I pulled that map into a separate file, I moved the function to that file and it's not inlined anymore.